src/AutoRest.CSharp/LowLevel/Generation/LowLevelExampleComposer.cs

```diff
@@ -0,0 +1,899 @@
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Text;
+using System.Text.Json;
+using Azure;
+using AutoRest.CSharp.Generation.Types;
+using AutoRest.CSharp.Input;
+using AutoRest.CSharp.Output.Builders;
+using AutoRest.CSharp.Output.Models;
+using AutoRest.CSharp.Output.Models.Shared;
+using AutoRest.CSharp.Output.Models.Types;
+using Azure.Core;
+
+namespace AutoRest.CSharp.Generation.Writers
+{
+    internal class LowLevelExampleComposer
+    {
+        private CodeWriter Writer { get; }
+        private string ClientTypeName { get; }
+        private BuildContext<LowLevelOutputLibrary> Context { get; }
+
+        public LowLevelExampleComposer(CodeWriter writer, string clientTypeName, BuildContext<LowLevelOutputLibrary> context)
+        {
+            Writer = writer;
+            ClientTypeName = clientTypeName;
+            Context = context;
+        }
+
+        public void Write(LowLevelClientMethod clientMethod, bool async)
+        {
+            var methodSignature = clientMethod.Signature.WithAsync(async);
+            var operationSchema = clientMethod.OperationSchemas;
+
+            var examples = new List<string>();
+
+            if (HasNoCustomInput(methodSignature.Parameters))
+            {
+                examples.Add(GenerateExampleWithoutParameter(clientMethod, methodSignature.Name, async, true));
+            }
+            else if (HasOptionalInputValue(methodSignature.Parameters, operationSchema.RequestBodySchema))
+            {
+                if (AreAllParametersOptional(methodSignature.Parameters))
+                {
+                    examples.Add(GenerateExampleWithoutParameter(clientMethod, methodSignature.Name, async, false));
+                }
+                else if (operationSchema.RequestBodySchema != null && HasRequiredAndWritablePropertyFromTop(operationSchema.RequestBodySchema))
+                {
+                    examples.Add(GenerateExampleWithParametersAndRequestContent(clientMethod, methodSignature.Name, async, false));
+                }
+                else
+                {
+                    examples.Add(GenerateExampleWithoutRequestContent(clientMethod, methodSignature.Name, async));
+                }
+                examples.Add(GenerateExampleWithParametersAndRequestContent(clientMethod, methodSignature.Name, async, true));
+            }
+            else
+            {
+                examples.Add(GenerateExampleWithRequiredParameters(clientMethod, methodSignature.Name, async));
+            }
+
+            Writer.WriteXmlDocumentation("example", $"{string.Join(Environment.NewLine, examples)}");
+        }
+
+        private bool HasNoCustomInput(IReadOnlyList<Parameter> parameters)
+        {
+            return parameters.Count() <= 1; // `RequestContext = null` is excluded
```

Comment:
I think this will throw exception if there is no parameter (though not possible, since every protocol method should end with `RequestContext`). I'll change it to `parameters.Where(p => p.Name != "RequestContext").Count == 0`.

Reply:
After change, I think the comment is useless.